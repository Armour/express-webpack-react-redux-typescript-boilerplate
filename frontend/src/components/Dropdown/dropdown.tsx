import React from 'react';
import { Translation } from 'react-i18next';
import { Link } from 'react-router-dom';

interface IDropdownProps {
  id: string;
  dropdownLists: string[];
}

const dropdownConfig: Partial<M.DropdownOptions> = {
  coverTrigger: false,
};

export class Dropdown extends React.Component<IDropdownProps> {
  public componentDidMount() {
    const elems = document.querySelectorAll('.dropdown-button');
    M.Dropdown.init(elems, dropdownConfig);
  }

  public render() {
    return (
      <Translation ns='common'>
        {(t) => (
          <ul id={this.props.id} className='dropdown-content'>
            {this.props.dropdownLists.map((key) =>
              <li key={key}><Link to={`/${key}`}>{t('dropdown.' + key)}</Link></li>,
            )}
          </ul>
        )}
      </Translation>
    );
  }
}

export default Dropdown;
