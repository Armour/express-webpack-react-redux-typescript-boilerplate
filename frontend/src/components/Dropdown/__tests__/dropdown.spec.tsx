import 'materialize-css';
import React, { Fragment } from 'react';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';

import { Dropdown } from '../dropdown';

describe('Dropdown', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <BrowserRouter>
        <Fragment>
          <Dropdown id='id' dropdownLists={['dropdown']} />,
        </Fragment>
      </BrowserRouter>,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
