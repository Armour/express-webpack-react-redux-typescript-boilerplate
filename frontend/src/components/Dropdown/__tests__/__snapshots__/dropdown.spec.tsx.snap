// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should renders correctly 1`] = `
Array [
  <ul
    className="dropdown-content"
    id="id"
  >
    <li>
      <a
        href="/dropdown"
        onClick={[Function]}
      >
        dropdown.dropdown
      </a>
    </li>
  </ul>,
  ",",
]
`;
